src/transaction/log_lsa.hpp

```diff
@@ -64,6 +64,9 @@ constexpr std::int64_t NULL_LOG_PAGEID = -1;
 constexpr std::int16_t NULL_LOG_OFFSET = -1;
 constexpr log_lsa NULL_LSA { NULL_LOG_PAGEID, NULL_LOG_OFFSET };
 
+constexpr std::int64_t MAX_LOG_LSA_PAGEID = (static_cast<std::int64_t> (1u) << (48 - 1)) - 1;
+constexpr std::int16_t MAX_LOG_LSA_OFFSET = (static_cast<std::int16_t> (1u) << (16 - 1)) - 1;
```

Comment:
- 16 times it does overflow; 15 times it does not;
- `SHRT_MAX` has the same value as the one generated by the shift expression;
- nothing against changing for `MAX_LOG_LSA_OFFSET` but for `MAX_LOG_LSA_PAGEID` it would have to remain explicit as there is no built in expression
- I thought that having them laid out explicitly is nicer and more explicit;
- I will add a comment though